bitratechecker.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python2
+
+from __future__ import print_function
+import sys
+import os
+import glob
+import subprocess
+import json
+import argparse
+
+class Mp3DownloadAnalyzer(object):
+    def __init__(self, mp3_files):
+
+        if not len(mp3_files) > 0:
+            raise ValueError
+
+        self.__name = []
+        self.__bitrate = []
+        self.__mode = []
+        self.__library = []
+        self.__encoder = []
+
+        self.__get_media_info(mp3_files)
+        self.__guess_encoder(mp3_files)
+
+        self.__same_bitrate = len(set(self.__bitrate)) == 1
+        self.__same_library = len(set(self.__library)) == 1
+        self.__same_encoder = len(set(self.__encoder)) == 1
+        self.__all_cbr = self.__mode.count("CBR") == len(self.__mode)
+        self.__all_vbr = self.__mode.count("VBR") == len(self.__mode)
+
+
+    def __get_media_info(self, mp3_files):
+        try:
+            raw_json = subprocess.check_output(["mediainfo", "--output=JSON"] + mp3_files)
+        except subprocess.CalledProcessError:
+            raise EnvironmentError
+
+        media = json.loads(raw_json)
+
+        for m in media:
+            self.__name.append(os.path.basename(str(m["media"]["@ref"])))
+
+            try:
+                self.__bitrate.append(int(m["media"]["track"][0]["OverallBitRate"]))
+            except KeyError:
+                self.__bitrate.append(-1)
+
+            try:
+                self.__mode.append(str(m["media"]["track"][0]["OverallBitRate_Mode"]))
+            except KeyError:
+                self.__mode.append("unknown")
+
+            try:
+                self.__library.append(str(m["media"]["track"][0]["Encoded_Library"].encode("utf-8").strip()))
+            except KeyError:
+                self.__library.append("unknown")
+            except AttributeError:
+                self.__library.append("unreadable")
+
+
+    def __guess_encoder(self, mp3_files):
+        for f in mp3_files:
+            try:
+                enc = subprocess.check_output(["mp3guessenc", "-n", f]).splitlines()[0]
+            except subprocess.CalledProcessError as e:
+                # mp3guessenc returns different statuses for different encoders - not just zero on success
+                # so we actually get meaningful output here in except block
+                enc = e.output.splitlines()[0]
+
+            self.__encoder.append(enc)
+
+
+    def print_formatted(self):
```

Comment:
Maybe it would be better to implement `__str__` instead?

Reply:
Yes, that's good idea

Reply:
Done.